Library/Backends/ZendEngine2/Backend.php

```diff
@@ -242,8 +242,19 @@ public function onPreCompile($method, CompilationContext $context)
             $codePrinter->increaseLevel();
             $codePrinter->output('{');
             $codePrinter->increaseLevel();
-            $codePrinter->output('zval *this_ptr = NULL;');
-            $codePrinter->output('ZEPHIR_CREATE_OBJECT(this_ptr, class_type);');
+            $codePrinter->output('zval zthis       = zval_used_for_init;');
+            $codePrinter->output('zval *this_ptr   = &zthis;');
+            $codePrinter->output('zend_object* obj = ecalloc(1, sizeof(zend_object));');
+            $codePrinter->output('zend_object_value retval;');
+            $codePrinter->outputBlankLine();
+            $codePrinter->output('zend_object_std_init(obj, class_type TSRMLS_CC);');
+            $codePrinter->output('object_properties_init(obj, class_type);');
+            $codePrinter->output('retval.handle   = zend_objects_store_put(obj, (zend_objects_store_dtor_t)zend_objects_destroy_object, zephir_free_object_storage, NULL TSRMLS_CC);');
+            $codePrinter->output('retval.handlers = zend_get_std_object_handlers();');
+            $codePrinter->outputBlankLine();
+            $codePrinter->output('Z_TYPE(zthis)   = IS_OBJECT;');
+            $codePrinter->output('Z_OBJVAL(zthis) = retval;');
+            $codePrinter->outputBlankLine();
```

Comment:
No, I suggest to use something like `templates/ZendEngine2/zephir_init_properties.tpl`. I'm still not sure. Just wondering your opinion